lib/puppet/type/cisco_stp_global.rb

```diff
@@ -0,0 +1,755 @@
+# Manages the Cisco Spanning-tree Global configuration resource.
+#
+# Jan 2016
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+begin
+  require 'puppet_x/cisco/cmnutils'
```

Comment:
I know but if I remove it from this file and put this in nxapi.rb, I am getting error(see below). Not sure why but it works only if this is in the type rather than provider. We can debug this when you have some time.

Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Resource Statement, Could not autoload puppet/type/cisco_stp_global: Could not autoload puppet/provider/cisco_stp_global/nxapi: no such file to load -- /etc/puppetlabs/code/environments/production/modules/ciscopuppet/lib/puppet/puppet_x/cisco/cmnutils at /etc/puppetlabs/code/environments/production/manifests/cisco/demo_stp_vlan.pp:47:3 on node saipuppetagent.cisco.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run